swift-ui-base/Common/View/TextField/Model/TextFieldData.swift

```diff
@@ -11,16 +11,16 @@ import Foundation
 final class TextFieldData: ObservableObject {
   var value: String = "" {
     didSet {
-      hasTyped = hasTyped || !value.isEmpty
       validate()
     }
   }
+  
   var validationType: ValidationType
   var errorMessage: String
   var title: String
   var isSecure = false
-  var hasTyped = false
-  var isValid = true
+
+  var isValid = true//@Published var isValid = true
```

Comment:
remove comment if not used. Maybe open an issue to make these classes Combine friendly.

Reply:
Agree, I only create this PR to change the way the textfield behaves based on states not to modify the model behind. That should be addressed in a different PR

Reply:
how's it not combine friendly?